src/client/replication_ddl_client.cpp

```diff
@@ -226,8 +226,11 @@ dsn::error_code replication_ddl_client::drop_app(const std::string &app_name, in
 
 dsn::error_code replication_ddl_client::recall_app(int32_t app_id, const std::string &new_app_name)
 {
-    RETURN_EC_NOT_OK_MSG(
-        validate_app_name(new_app_name), "invalid new_app_name: '{}'", new_app_name);
+    if (!std::all_of(new_app_name.cbegin(),
+                     new_app_name.cend(),
+                     (bool (*)(int))replication_ddl_client::valid_app_char)) {
```

Comment:
valid_app_char() has been removed.

It would be better to add a new boolean parameter for validate_app_name() to indicate whether empty is allowed or not.